### PR TITLE
[docs] Start a version log in the documentation.

### PR DESCRIPTION
**Pull request by dHannasch**
## Why are these changes needed?

This starts a version log in the documentation. In particular, it starts a log of changes in 1.1.0.
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by richardliaw** [comment]
Hmm, can you try building the docs to catch errors?

File: doc/source/index.rst [hunk 1]
```diff
@@ -198,6 +198,7 @@ Papers
    ray-overview/index.rst
    ray-libraries.rst
    installation.rst
+   versions.rst
```
Comment (richardliaw) [inline doc/source/index.rst:201]: can we actually move this to the bottom for now? (in the Contributing section)

**Comment by dHannasch** [comment]
Does this look okay? Obviously this is out of date since a bunch more things have been merged, for 1.1.0 this currently only has the first few merged PRs post-1.0.1, but I'm a bit reluctant to add more to this branch since this actually got a green checkmark from Travis and that happens so rarely.

**Comment by richardliaw** [comment]
Hmm I think I will close this for now as we can just link to the release notes on Github instead.

**Pull request closed without merging**